examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap

```diff
@@ -298,10 +303,11 @@ layout(std430, binding=1) buffer blocked_result{
 };
 
 void main(){
-  for(int i = 0; i < MAT_SIZE; i++){
-    for(int j = 0; j < MAT_SIZE; j++){
-        // Converts the 4x4 array of 4x4 matrices to a 16x16 array
-        res[i][j] = blocked_res[i/KERNEL_SIZE][j/KERNEL_SIZE][j%KERNEL_SIZE][i%KERNEL_SIZE];
+  uint row = uint(gl_WorkGroupID[0]), col = uint(gl_WorkGroupID[1]);
+  for(int i = 0; i < KERNEL_SIZE; i++){
+    for(int j = 0; j < KERNEL_SIZE; j++){
+        // Converts the 4x4 array of 4x4 matrices to a 16x16array
+        res[row * KERNEL_SIZE + i][col * KERNEL_SIZE + j] = blocked_res[row][col][j][i];
```

Comment:
This looks correct.

An alternative is to use a 16x16 set of workgroups and to have each invocation take care of a single submatrix element.

I'll leave it to you to decide what's best.

Reply:
Same, changed the workgroups to 16x16 and replaced the nested for-loop by:
```res[row][col] = blocked_res[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE];```